applications/MappingApplication/custom_mappers/nearest_neighbor_mapper_iga.cpp

```diff
@@ -0,0 +1,118 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:    Juan Ignacio Camarotti
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "nearest_neighbor_mapper_iga.h"
+#include "mapping_application_variables.h"
+
+namespace Kratos {
+
+void NearestNeighborInterfaceInfoIGA::ProcessSearchResult(const InterfaceObject& rInterfaceObject)
+{
+    SetLocalSearchWasSuccessful();
+
+    const double neighbor_distance = MapperUtilities::ComputeDistance(this->Coordinates(), rInterfaceObject.Coordinates());
+
+    const auto& shape_function_values = rInterfaceObject.pGetBaseGeometry()->ShapeFunctionsValues();
+    const std::size_t num_values = shape_function_values.size2();
+
+    if (neighbor_distance < mNearestNeighborDistance) {
+        mNearestNeighborDistance = neighbor_distance;
+        mNumberOfNearestNeighbors = 1;
+
+        mShapeFunctionValues.clear();
+        mNearestNeighborId.clear();
+
+        // Reserve space for the vectors
+        mShapeFunctionValues.reserve(num_values);
+        mNearestNeighborId.reserve(num_values);
```

Comment:
you're appending to possibly non-empty containers
```suggestion
        // Reserve space for the vectors
        mShapeFunctionValues.reserve(mShapeFunctionValues.size() + num_values);
        mNearestNeighborId.reserve(mNearestNeighborId.size() + num_values);
```